notary-sora-integration-test/src/integration-test/kotlin/integration/sora/SoraIntegrationTest.kt

```diff
@@ -0,0 +1,159 @@
+package integration.sora
+
+import integration.helper.IrohaIntegrationHelperUtil
+import jp.co.soramitsu.iroha.ModelCrypto
+import model.IrohaCredential
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import sidechain.iroha.util.ModelUtil
+import util.getRandomString
+import kotlin.test.assertEquals
+
+/**
+ * This test case tests Sora integration.
+ * All accounts are created and have permissions.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SoraIntegrationTest {
+
+    val integrationHelper = IrohaIntegrationHelperUtil()
+
+    val domain = "sora"
+    val xorAsset = "xor#$domain"
+
+    init {
+        System.loadLibrary("irohajava")
+    }
+
+    /**
+     * Sora user can query her/his balance.
+     * @given a client with 1334 xor on balance
+     * @when the client queries the balance
+     * @then the actual balance of 1334 xor is returned
+     */
+    @Test
+    fun getXorBalanceTest() {
+        val clientName = String.getRandomString(9)
+        val clientId = "$clientName@$domain"
+
+        val keypairAlice = ModelCrypto().generateKeypair()
+        println("pub")
+        println(keypairAlice.publicKey().hex())
+        println("priv")
+        println(keypairAlice.privateKey().hex())
+
+        val clientAliceCredential = IrohaCredential(clientId, keypairAlice)
+
+        integrationHelper.createAccount(clientName, domain, keypairAlice.publicKey())
+
+        assertEquals(
+            "0",
+            integrationHelper.getAccountAssets(clientAliceCredential, clientId).getOrDefault(xorAsset, "0")
```

Comment:
`getAccountAssetBalance` should be used I think

Reply:
no, client should sign the tx to check permissions are correct.

Reply:
but the difference is only in `.map { it.getOrDefault(assetId, "0") }`, isn't it?